core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/UnaryTupleOperator.java

```diff
@@ -90,13 +90,17 @@ public void replaceChildNode(QueryModelNode current, QueryModelNode replacement)
 	}
 
 	@Override
-	public boolean equals(Object other) {
-		if (other instanceof UnaryTupleOperator) {
-			UnaryTupleOperator o = (UnaryTupleOperator) other;
-			return arg.equals(o.getArg());
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
 		}
+		if (!(o instanceof UnaryTupleOperator)) {
+			return false;
+		}
+
+		UnaryTupleOperator that = (UnaryTupleOperator) o;
 
-		return false;
+		return arg.equals(that.arg);
```

Comment:
I think you may need a null-check here as well - perhaps use `Objects.equals(arg, that.arg)`?

Reply:
And if we're being consistent, should this class `hashCode` also do a null check? 

Fwiw I think that generally the way we handle this in these classes is a little inconsistent. We have a no-arg constructor (which effectively allows the `arg` to be `null`), but then the setter has assertions to ensure it can not be set to `null`. It would probably be cleaner to come down one way or the other. But not necessarily in this PR, happy with a quick fix :)

Reply:
I took a look at `UnaryTupleOperator` and I couldn't find any scenarios where `arg` ends up being `null`. The values that I see for `arg` are all some form of a query body, and even and empty query body won't result in `null`. For `SELECT * WHERE { }` we use a `SingletonSet` as the query body. 

The issues with `UnaryValueOperator` is that it's not supposed to handle multiple arguments, but it still needs to handle `*`, even the SPARQL spec lists it as a special case. Our approach to handling `*` in `SELECT * WHERE {?a a ?type}` is to expand `*` into `?a ?type`. That's intentionally not allowed with the `UnaryValueOperator` because it's not allowed to have multiple arguments with something like `COUNT`.

Maybe for 5.0.0 we could consider two things for the algebra:
- Introduce something to represent `*` in `UnaryValueOperator` (maybe a new class or just a constant).
- Remove all the empty constructors from the classes that actually do require arguments.